data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java

```diff
@@ -111,12 +113,20 @@ private void buildPipelineFromConfiguration(
         LOG.info("Building [{}] as source component for the pipeline [{}]", pluginSetting.getName(), sourcePipelineName);
         final Optional<String> pipelineNameOptional = getPipelineNameIfPipelineType(pluginSetting);
         if (pipelineNameOptional.isPresent()) { //update to ifPresentOrElse when using JDK9
+            final String connectedPipeline = pipelineNameOptional.get();
             if (!sourceConnectorMap.containsKey(sourcePipelineName)) {
                 LOG.info("Source of pipeline [{}] requires building of pipeline [{}]", sourcePipelineName,
-                        pipelineNameOptional.get());
+                        connectedPipeline);
                 //Build connected pipeline for the pipeline connector to be available
+                //Building like below sometimes yields multiple runs if the pipeline building fails before sink
+                //creation. except for running the creation again, it will not harm anything - TODO Fix this
                 buildPipelineFromConfiguration(pipelineNameOptional.get(), pipelineConfigurationMap, pipelineMap);
             }
+            if (!pipelineMap.containsKey(connectedPipeline)) {
+                LOG.info("Connected Pipeline [{}] failed to build, Failing building source for [{}]",
```

Comment:
log.warn at least?

Reply:
my bad - it should have been LOG.error; Erick will fix it in following PR